File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack st = new Stack<Character>();
        StringBuilder ans = new StringBuilder();
        for(char ch:s.toCharArray()){
            if(ch == '('){
                if(!st.isEmpty()){
                    ans= ans.append(ch);
                }
                st.push(ch);
            }
            else{
                st.pop();
                 if(!st.isEmpty()){
                    ans= ans.append(ch);
                }
            }
        }
      return ans.toString();
    }
}

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        stack<char> st ;
        string ans= "";
        for(char sh:s){
            if(sh=='('){
                if(!st.empty()){
                    ans+=sh;
                }
                st.push(sh);
            }else{
                st.pop();
                if(!st.empty()){
                    ans+=sh;
                }
            }
            
        }
        return ans;
    }
};